firebase-firestore/src/main/java/com/google/firebase/firestore/core/FirestoreClient.java

```diff
@@ -127,6 +114,21 @@ public FirestoreClient(
             throw new RuntimeException(e);
           }
         });
+
+    credentialsProvider.setChangeListener(
+        (User user) -> {
+          if (initialized.compareAndSet(false, true)) {
+            hardAssert(!firstUser.getTask().isComplete(), "Already fulfilled first user task");
+            firstUser.setResult(user);
+          } else {
+            // this fires twoce and sync engine isnt' initialized
+            asyncQueue.enqueueAndForget(
+                () -> {
+                  Logger.debug(LOG_TAG, "Credential changed. Current user: %s", user.getUid());
+                  syncEngine.handleCredentialChange(user);
```

Comment:
Optional: perhaps assert that `syncEngine` is not null? Or would this be trying too hard to protect against past problems?

Reply:
I don't think it will hurt. Done.